planner/core/task.go

```diff
@@ -1115,14 +1115,18 @@ func (p *PhysicalTopN) canExpressionConvertedToPB(storeTp kv.StoreType) bool {
 
 // containVirtualColumn checks whether TopN.ByItems contains virtual generated columns.
 func (p *PhysicalTopN) containVirtualColumn(tCols []*expression.Column) bool {
+	tColMap := make(map[int64]*expression.Column, len(tCols))
```

Comment:
```suggestion
	tColSet := make(map[int64]struct{}, len(tCols))
```

It should be set. if it is map, the value is useless.